File: lab3-24.04.20/abandoned/math/MathFunctionService.java
package math;

import java.util.HashSet;
import static java.lang.Math.*;

public class MathFunctionService {

    public enum SolveMethods {
        NEWTON_POLYNOMIAL { @Override public String toString() { return "Интерполирование многочленом Ньютона"; }},
        LAGRANGE_POLYNOMIAL { @Override public String toString() { return "Интерполирование многочленом Лагранжа"; }},
        CUBIC_SPLINE { @Override public String toString() { return "Интерполирование кубическими сплайнами"; }}
    }

    //TODO: норм объекты
    private static final Object solver;
    public static final HashSet<MathFunction> equations;

    static {
        solver = new InterpolationSolver();
        equations = new HashSet<>();
        equations.add(new MathFunction() {
            @Override public double func(double x) { return sin(x); }
            @Override public String toString() { return "y = sin(x)"; }
        });
    }

    public static InterpolationAnswer solve() {
        //TODO: выбор метода
        return new InterpolationAnswer();
    }
}

File: lab3-24.04.20/src/math/MathFunctionService.kt
package math

import kotlin.math.*

//TODO: проверить документацию
/**
 * Предоставляет единую точку доступа к экземпляру класса {@see NonLinearEquationSolver}.
 * @author Артемий Кульбако.
 * @version 1.4
 */
class MathFunctionService {

    /**
     * Константы, определяющие методы решения системы нелинейных уравнений.
     * @property NEWTON_POLYNOMIAL метод половинного деления.
     * @property LAGRANGE_POLYNOMIAL метод касательных (метод Ньютона).
     * @property CUBIC_SPLINE метод простых итераций.
     */
    enum class SolveMethods {
        NEWTON_POLYNOMIAL { override fun toString() = "Интерполирование многочленом Ньютона" },
        LAGRANGE_POLYNOMIAL { override fun toString() = "Интерполирование многочленом Лагранжа" },
        CUBIC_SPLINE { override fun toString() = "Интерполирование кубическими сплайнами" }
    }

    /**
     * @property equations нелинейные уравнения с одной неизвестной переменной.
     */
    companion object {

        private val solver = InterpolationSolver()
        val equations: Set<MathFunction>

        init {
            equations = setOf(
                object : MathFunction {
                    override fun func(vararg x: Double) = x[0] / 2 - 3
                    override fun toString() = "x/2 - 3 = y"   //ПРОВЕРЕНА
                }
            )
        }

        fun solve() { }
    }
}